tempodb/backend/tenantindex.go

```diff
@@ -3,22 +3,19 @@ package backend
 import (
 	"bytes"
 	"encoding/json"
+	"fmt"
 	"time"
 
+	proto "github.com/gogo/protobuf/proto"
 	"github.com/klauspost/compress/gzip"
+	"github.com/klauspost/compress/zstd"
 )
 
 const (
 	internalFilename = "index.json"
 )
 
-// TenantIndex holds a list of all metas and compacted metas for a given tenant
-// it is probably stored in /<tenantid>/blockindex.json.gz as a gzipped json file
-type TenantIndex struct {
-	CreatedAt     time.Time             `json:"created_at"`
-	Meta          []*BlockMeta          `json:"meta"`
-	CompactedMeta []*CompactedBlockMeta `json:"compacted"`
-}
+var _ proto.Message = &TenantIndex{}
```

Comment:
+1 for checking the interface.  Usually it's done with a nil pointer to avoid allocating the object.
```suggestion
var _ proto.Message = (*TenantIndex)(nil)
```